### ISSUE TITLE
Validations not happening for the properties present in the node_types

### ISSUE THREAD

**Issue by abhishek-kumar0409** [tliron/puccini#17]
Hi Tal,

While playing with puccini, I also discovered that no validations are happening for the fields present in properties section. I'm referring about the validation of the the default values against the valid_values defined in the constraints.  Another issue observed was for the property "vnfm_info". The constraints field should be defined or indented within the scope of entry_schema. Should puccini be responsible for throwing the error or it's not required at all assuming that its a typo-error from the user's end?

    node_types:
       MyCompany.SunshineDB.1_0.1_0:
         derived_from: tosca.nodes.nfv.VNF
         properties:
            descriptor_id:
                type: string
                constraints: [ valid_values: [ b1bb0ce7-ebca-4fa7-95ed-4840d70a1177 ]]
                default: b1bb0ce7-ebca-4fa7-95ed-4840d70a1177
            provider:
                type: string
                constraints: [ valid_values: [ MyCompany] ]
                default: MyCompany
           vnfm_info:
                type: list
                entry_schema:
                   type: string
                   constraints: [ valid_values: [MyCompany ] ]

You can test with the VNFDs that I shared in the comments section of the issue #12 . Please confirm if this is a valid bug or not.

**Comment by tliron** [comment]
Are you running this with `puccini-tosca compile --coerce`? Otherwise constraints are not checked.

**Comment by abhishek-kumar0409** [comment]
Hi Tal,

Yes , I am running with Puccini tosca-compile command because  the support
for quirks are available only to puccini-tosca compile command...

Best regards,
Abhishek

**Comment by tliron** [comment]
With `--coerce`?

**Comment by abhishek-kumar0409** [comment]
Yes , without --coerce. Should I run with --coerce ?

**Comment by tliron** [comment]
Yes. :) As I said, that's how constraints are checked. :) The compiler does not in itself run any intrinsic functions or check constraints. You can either use `--coerce` for that or pipe to `puccini-js exec tosca.coerce` (does almost the same thing).

**Comment by abhishek-kumar0409** [comment]
Oops... Thanks for the update. I will test and let you know about that..😊

**Comment by abhishek-kumar0409** [comment]
Hi Tal,
You can close this issue. With your resolution, I am able to validate the node_types. Thanks much.
